app/screens/send-bitcoin-screen/payment-details/onchain-payment-details.ts

```diff
@@ -0,0 +1,291 @@
+        variables: {
+          walletId: sendingWalletDescriptor.id,
+          address,
+          amount: settlementAmount.amount,
+        },
+      })
+
+      const rawAmount = data?.onChainTxFee.amount
+      const amount =
+        typeof rawAmount === "number"
+          ? {
+              amount: rawAmount,
+              currency: sendingWalletDescriptor.currency,
+            }
+          : rawAmount
+
+      return {
+        amount,
+      }
+    }
+  }
+
+  const setAmount: SetAmount<T> | null = (newUnitOfAccountAmount) => {
+    return CreateNoAmountOnchainPaymentDetails({
+      ...params,
+      unitOfAccountAmount: newUnitOfAccountAmount,
+    })
+  }
+
+  const setMemo: SetMemo<T> | null = destinationSpecifiedMemo
+    ? null
+    : (newMemo) => {
+        return {
+          ...CreateNoAmountOnchainPaymentDetails({
+            ...params,
+            senderSpecifiedMemo: newMemo,
+          }),
+          getFee,
+        }
+      }
+
+  const setConvertPaymentAmount = (newConvertPaymentAmount: ConvertPaymentAmount) => {
+    return CreateNoAmountOnchainPaymentDetails({
+      ...params,
+      convertPaymentAmount: newConvertPaymentAmount,
+    })
+  }
+
+  const setSendingWalletDescriptor: SetSendingWalletDescriptor<T> = (
+    newSendingWalletDescriptor,
+  ) => {
+    return CreateNoAmountOnchainPaymentDetails({
+      ...params,
+      sendingWalletDescriptor: newSendingWalletDescriptor,
+    })
+  }
+
+  const setUnitOfAccount: SetUnitOfAccount<T> = (newUnitOfAccount) => {
+    return CreateNoAmountOnchainPaymentDetails({
+      ...params,
+      unitOfAccountAmount:
+        unitOfAccountAmount &&
+        convertPaymentAmount(unitOfAccountAmount, newUnitOfAccount),
+    })
+  }
+
+  return {
+    destination: address,
+    destinationSpecifiedAmount: null,
+    settlementAmount,
+    settlementAmountIsEstimated:
+      sendingWalletDescriptor && sendingWalletDescriptor.currency !== "BTC",
+    unitOfAccountAmount,
+    sendingWalletDescriptor,
+    memo,
+    paymentType: "onchain",
+    setSendingWalletDescriptor,
+    setUnitOfAccount,
+    convertPaymentAmount,
+    setConvertPaymentAmount,
+    setMemo,
+    setAmount,
+    sendPayment,
+    getFee,
+  }
+}
+
+export type CreateAmountOnchainPaymentDetailsParams<T extends WalletCurrency> = {
+  address: string
+  destinationSpecifiedAmount: BtcPaymentAmount
+  unitOfAccount: WalletCurrency
+} & BaseCreatePaymentDetailsParams<T>
+
+export const CreateAmountOnchainPaymentDetails = <T extends WalletCurrency>(
+  params: CreateAmountOnchainPaymentDetailsParams<T>,
+): PaymentDetail<T> => {
+  const {
+    unitOfAccount,
+    destinationSpecifiedAmount,
+    convertPaymentAmount,
+    sendingWalletDescriptor,
+    destinationSpecifiedMemo,
+    senderSpecifiedMemo,
+    address,
+  } = params
+
+  if (sendingWalletDescriptor && sendingWalletDescriptor.currency !== "BTC") {
+    throw new Error("Onchain payments are only supported for BTC wallets")
```

Comment:
it's no longer the case. we could use this PR to add onchain USD payment as well?

Reply:
Yeah we can do that

Reply:
I will hold off on USD for now, as there is work to be done on the backend to properly support amount usd invoices.